src/emmeans.jl

```diff
@@ -73,13 +82,27 @@ function emmeans(model::RegressionModel; eff_col=nothing, err_col=:err,
     if !isnothing(dof)
         result[!, :dof] .= _dof(dof, model)
     end
+    if !isnothing(ci_level)
+        # don't include this in the above if because
+        # we don't want to potentially add a dof column if there is no CI
+        if isnothing(dof)
+            # fall back to z value
+            result[!, :dof] .= Inf
+        end
+        # divide by two for twosided
+        scale = abs.(quantile.(TDist.(result[!, :dof]), (1 - ci_level) / 2))
+        result[!, lower_col] .= result[!, eff_col] .- result[!, err_col] .* scale
+        result[!, upper_col] .= result[!, eff_col] .+ result[!, err_col] .* scale
```

Comment:
Rather than duplicating this between `emmeans` and `empairs`, you could factor it out into a separate function, e.g.:
```julia
function _ci!(df, level; eff_col, err_col, dof, lower_col, upper_col)
    if isnothing(dof)
        dof = "dof"
        result[!, dof] .= Inf
    end
    transform!(df, [eff_col, err_col, dof] => ByRow() do eff, err, ν
                   scale = quantile(TDist(ν), (1 + level) / 2)
                   lower = eff - scale * err
                   upper = eff + scale * err
                   return (lower, upper)
               end => [lower_col, upper_col])
    return df
end
```
(The use of `transform!` here avoids the need to allocate `scale` as an array, but you could obviously also leave it structured as you have it.)